src/wizard/wizard-dialog.ts

```diff
@@ -327,93 +328,74 @@ export class ESPHomeWizardDialog extends LitElement {
-        <mwc-radio
-          name="board"
-          .value=${BOARD_GENERIC_ESP32C3}
-          @click=${this._handlePickBoardRadio}
-          ?checked=${this._board === BOARD_GENERIC_ESP32C3}
-        ></mwc-radio>
-      </mwc-formfield>
-
-      <mwc-formfield label="ESP8266">
-        <mwc-radio
-          name="board"
-          .value=${BOARD_GENERIC_ESP8266}
-          @click=${this._handlePickBoardRadio}
-          ?checked=${this._board === BOARD_GENERIC_ESP8266}
-        ></mwc-radio>
+      <mwc-list class="platforms">
+        ${Object.keys(platforms).map(
+          (key) => html`
+            <mwc-list-item
+              hasMeta
+              .platform=${key}
+              @click=${this._handlePickPlatformClick}
+            >
+              <span>${platforms[key]}</span>
+              ${metaChevronRight}
+            </mwc-list-item>
+          `
+        )}
+      </mwc-list>
+
+      <mwc-formfield slot="secondaryAction" label="Use recommended settings">
+        <mwc-checkbox
+          name="use-recommended"
+          @change=${this._handleUseRecommendedCheckbox}
+          ?checked=${this._useRecommended}
+        ></mwc-checkbox>
       </mwc-formfield>
 
-      <mwc-formfield label="Raspberry Pi Pico W">
-        <mwc-radio
-          name="board"
-          .value=${BOARD_RPI_PICO_W}
-          @click=${this._handlePickBoardRadio}
-          ?checked=${this._board === BOARD_RPI_PICO_W}
-        ></mwc-radio>
-      </mwc-formfield>
+      <mwc-button
+        no-attention
+        slot="primaryAction"
+        dialogAction="close"
+        label="Cancel"
+      ></mwc-button>
+    `;
+  }
 
-      <mwc-formfield label="Pick specific board">
-        <mwc-radio
-          name="board"
-          value="CUSTOM"
-          @click=${this._handlePickBoardRadio}
-          ?checked=${this._board === "CUSTOM"}
-        ></mwc-radio>
-      </mwc-formfield>
-      ${this._board !== "CUSTOM"
-        ? ""
+  private _renderPickBoard() {
+    return html`
+      ${this._error ? html`<div class="error">${this._error}</div>` : ""}
+      ${this._busy
+        ? html`<div>Loading board list...</div>`
         : html`
-            <div class="formfield-extra">
-              <select @change=${this._handlePickBoardCustom}>
-                ${boardSelectOptions(this._supportedBoards)}
-              </select>
-            </div>
+            <select
+              @change=${this._handlePickBoardSelect}
+              size="15"
+              style="width: 100%;"
+            >
+              ${boardSelectOptions(this._supportedBoards, this._defaultBoard)}
+            </select>
           `}
-      <div>
-        Pick a custom board if the default targets don't work or if you want to
-        use the pin numbers printed on the device in your configuration.
-      </div>
 
       <mwc-button
         slot="primaryAction"
         label="Next"
         @click=${this._handlePickBoardSubmit}
+        ?disabled=${this._board === null}
```

Comment:
Since `defaultBoard` is `string | null`, there may be a case that a platform has no default board. In that case, we don't allow the user to confirm his choice without making a choice first.